src/scripts/fetchHistoricalMultiSigEventsOptimized.ts

```diff
@@ -0,0 +1,355 @@
+/* eslint no-console: 0 */
+
+import 'dotenv/config';
+
+import { eq } from 'drizzle-orm';
+import {
+  getProposalApprovedEvents,
+  getProposalDequeuedEvents,
+} from 'src/app/governance/proposalEvents';
+import database from 'src/config/database';
+import { approvalsTable } from 'src/db/schema';
+import fetchHistoricalMultiSigEventsAndSaveToDBProgressively from 'src/features/governance/fetchHistoricalMultiSigEventsAndSaveToDBProgressively';
+import updateApprovalsInDB from 'src/features/governance/updateApprovalsInDB';
+import { Chain, createPublicClient, http, PublicClient, Transport } from 'viem';
+import { celo } from 'viem/chains';
+
+// Block time changed from 5 seconds to 1 second at block 31056500
+const BLOCK_TIME_CHANGE_BLOCK = 31_056_500n;
+
+// New block times (1 second per block, after block 31056500)
+const TEN_DAYS_IN_BLOCKS_NEW = 864_000n;
+const THREE_DAYS_IN_BLOCKS_NEW = 259_200n; // Execution window: 3 days after last confirmation
+const BACKWARD_SEARCH_BLOCKS_NEW = 400_000n;
+
+// Old block times (5 seconds per block, before block 31056500)
+const TEN_DAYS_IN_BLOCKS_OLD = 172_800n; // 10 days * 86400 sec/day / 5 sec/block
+const SIX_DAYS_IN_BLOCKS_OLD = 103_680n; // 6 days * 86400 sec/day / 5 sec/block - execution window after last confirmation
+const BACKWARD_SEARCH_BLOCKS_OLD = 80_000n;
```

Comment:
not the biggest issue but it'd be more error proof if written like

```
const DAY = 24 * HOUR

const TEN_DAYS_IN_BLOCKS_OLD = 10 * DAY * BLOCKS_PER_DAY_OLD
```

or similar

won't block merging because of that though